frontend/src/Pages/RoomSettings.js

```diff
@@ -52,13 +40,27 @@ TableRows.propTypes = {
  */
 export default function RoomSettings() {
 
-  // Dummy data for tables
   const roomHeaderData = ['#', 'Room Alias', 'Room ID'];
-  const roomBodyData = [
-    {id: 1, roomAlias: "Matrix Cerberus", roomId: "!hMaAAyWsrBAMXEfXso:cadair.com"},
-    {id: 2, roomAlias: "GSoC Interns & Mentors 2022", roomId: "!auFsXYPeEIXfseVOEm:ergaster.org"},
-    {id: 3, roomAlias: "Element Web/Desktop", roomId: "!YTvKGNlinIzlkMTVRl:matrix.org"}
-  ];
+  const [roomBodyData, setRoomBodyData] = useState([]);
+
+  useEffect( () => {
+    async function fetchRoomData() {
+      const resp = await axios.get('api/users/room-list');
+      const data = resp.data.content;
+
+      let roomData = [], index=1;
+      for(let key in data){
+        roomData.push({
+          id: index,
+          roomAlias: data[key],
+          roomId: key
+        })
+        index += 1;
+      }
```

Comment:
This is just another way of writing this with fewer variables, up to you if you prefer the style:

- `Object.entries` allows you to pull out object key pairs as tuples.
- `map` allows you to reformat that data as a set of objects.

```suggestion
      const roomData = Object.entries(data).map(([roomId, roomAlias], id) => ({
        id,
        roomAlias,
        roomId,
      }));
```

Reply:
Thanks, for the suggestion.
I've added it in PR #39 

https://github.com/MatrixDoctors/matrix-cerberus/blob/9c831560771d3b62aee293fdfb1631a5b9c6265d/frontend/src/Pages/RoomSettings.js#L56-L62